nym-vpn-core/crates/nym-statistics-api-client/src/client.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright 2025 - Nym Technologies SA <contact@nymtech.net>
+// SPDX-License-Identifier: GPL-3.0-only
+
+use std::{fmt, net::SocketAddr, time::Duration};
+
+use nym_http_api_client::{ApiClient, HttpClientError, NO_PARAMS, PathSegments, UserAgent};
+use serde::{Serialize, de::DeserializeOwned};
+use url::Url;
+
+use crate::{
+    error::{Result, StatisticsApiClientError},
+    routes,
+};
+
+// requests can unfortunately take a long time over the mixnet
+pub(crate) const NYM_STATISTICS_API_TIMEOUT: Duration = Duration::from_secs(60);
+
+#[derive(Clone, Debug)]
+pub struct StatisticsApiClient {
+    inner: nym_http_api_client::Client,
+}
+
+impl StatisticsApiClient {
+    pub fn new(base_url: Url, user_agent: UserAgent) -> Result<Self> {
+        Self::new_with_resolver_overrides(base_url, user_agent, None)
+    }
+
+    pub fn new_with_resolver_overrides(
+        base_url: Url,
+        user_agent: UserAgent,
+        static_addresses: Option<&[SocketAddr]>,
+    ) -> Result<Self> {
+        nym_http_api_client::Client::builder(base_url.clone())
```

Comment:
@simonwicky maybe we could have a helper to avoid repeating the static addresses setup because I believe it's being used somewhere else too.

Reply:
On second though, we don't really need a resolver override for this, since we'll be connected to use it